File: backend/src/main/java/com/optimiza/clickbarber/repository/ServicoRepository.java
package com.optimiza.clickbarber.repository;

import com.optimiza.clickbarber.model.Servico;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface ServicoRepository extends JpaRepository<Servico, Long> {

    @Query("SELECT s FROM Barbearia b INNER JOIN Servico s ON b.idExterno = :idExternoBarbearia")
    List<Servico> findByIdExternoBarbearia(@Param("idExternoBarbearia") UUID idExternoBarbearia);

}

File: backend/src/main/java/com/optimiza/clickbarber/repository/ServicoRepository.kt
package com.optimiza.clickbarber.repository

import com.optimiza.clickbarber.model.servico.Servico
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ServicoRepository : JpaRepository<Servico?, Long?> {

    @Query("SELECT s FROM Servico s INNER JOIN Barbearia b ON b.id = s.barbearia.id WHERE b.idExterno = :idExternoBarbearia")
    fun findByIdExternoBarbearia(@Param("idExternoBarbearia") idExternoBarbearia: UUID?): List<Servico?>?

}